[changelog] Release 2023-03-03 (#16360)
conan-io/conan-center-index: docs/changelog.md

# Changelog

> ### 03-March-2023 - 11:05 CET

Author: ```suggestion
### 07-March-2023 - 17:05 CET

- [hotfix] Automatic merge skips git conflicts
- [hotfix] Conan V2 ready label depends on PR author

### 03-March-2023 - 11:05 CET
```

# Changelog

> ### 07-March-2023 - 17:05 CET

Author: ```suggestion
### 10-March-2023 - 13:15 CET

- [hotfix] No longer update conan_v2_ready_references.yml automatically
- [bugfix] Validate Green and clean PRs first when executing automatic merge CI job

### 07-March-2023 - 17:05 CET
```

- [hotfix] Automatic merge skips git conflicts

> - [hotfix] Conan V2 ready label depends on PR author

Reviewer: What does that mean?

Reviewer 2: We can label a PR as v2 required to ensure it will be updated :)

Authors can not set labels :/

Author: Only PRs which are labeled with conan-v2-ready and the author is a Conan team member, will be merged automatically. It's double check, because once the Bot opens the PR (it's a team member), CCI will mark with C3I Config label, plus, only team members are allowed to change files in .c3i folder. So when merging we just ensure that all rules were followed and no one removed a label, for instance.

Reviewer 2: > labeled with conan-v2-ready and the author is a Conan team member,

This sounds very confusing, why would we open a PR changing a recipe and then have it auto merge?
It should still need to be reviewed 😕 

is this the config list? or about recipes?

Author: it's only about [conan-v2-ready](https://github.com/conan-io/conan-center-index/blob/master/.c3i/conan_v2_ready_references.yml).

The bot will open a PR as soon as another PR passed and has been merged, but it needs to pass by Conan v1 AND Conan v2 AND the reference should not be listed in ready_references file yet.

So Automatic merge is a separated CI job, it does not know the context when a PR was created, so we need to ask all about from Github, if it passed, labels, green checks, ... 

As we are talking about `conan_v2_ready_references.yml`, the Automatic Merge will check the author, and labels. Then in case it matches, the PR will be merged automatically.

Very similar to Bump version, but only Conan team members can change conan_v2_ready_references.yml file. Like I said, it's a double check, because the Build CI job will not allow a non-team member opening a PR to .c3i and label it as c3i config.

Reviewer 2: Wearing my contributor hat, i have a bias towards thinking about recipes and my default assumption was this was about recipes.

```suggestion
- [hotfix] config label for "v2 ready" depends on PR author
```

I think would be more clear for others who are not in the weeds of the backend of the service :) https://github.com/conan-io/conan-center-index/labels/docs-conan2-ready

It's still prefixed docs and not c3i
